src/components/learner-credit-management/BudgetAssignmentsTable.jsx

```diff
@@ -13,46 +13,138 @@ const BudgetAssignmentsTable = ({
-    manualFilters
-    isLoading={isLoading}
-    defaultColumnValues={{ Filter: TableTextFilter }}
-    FilterStatusComponent={FilterStatus}
-    columns={[
-      {
-        Header: 'Assignment details',
-        accessor: 'assignmentDetails',
-        Cell: AssignmentDetailsTableCell,
-        disableSortBy: true,
-      },
-      {
-        Header: 'Amount',
-        Cell: ({ row }) => `-${formatPrice(row.original.contentQuantity / 100, { maximumFractionDigits: 0 })}`,
-        disableFilters: true,
-      },
-    ]}
-    initialTableOptions={{
-      getRowId: row => row?.uuid?.toString(),
-    }}
-    initialState={{
-      pageSize: PAGE_SIZE,
-      pageIndex: DEFAULT_PAGE,
-      sortBy: [],
-      filters: [],
-    }}
-    fetchData={fetchTableData}
-    data={tableData.results}
-    itemCount={tableData.count}
-    pageCount={tableData.numPages}
-    EmptyTableComponent={CustomDataTableEmptyState}
-  />
-);
+}) => {
+  const [rowsSelected, setRowsSelected] = useState(false);
+
+  const CancelAction = ({ selectedFlatRows, ...rest }) => (
+    // Here is access to the selectedFlatRows, isEntireTableSelected, tableInstance
+    <Button variant="danger" onClick={() => console.log('Cancel', selectedFlatRows, rest)}>
+      {`Cancel (${selectedFlatRows.length})`}
+    </Button>
+  );
+
+  const RemindAction = ({ selectedFlatRows, ...rest }) => (
+    <Button onClick={() => console.log('Remind', selectedFlatRows)}>
+      {`Remind (${selectedFlatRows.length})`}
+    </Button>
+  );
+  
+  const ExtraAction = ({ text, selectedFlatRows, ...rest }) => (
+    <Button variant="outline-primary" onClick={() => console.log(`Extra Action ${text}`, selectedFlatRows, rest)}>
+      {`Refresh`}
+    </Button>
+  );
+
+  const rowSelected = (rowsSelected) => {
+    if (Object.keys(rowsSelected).length > 1) {
+      setRowsSelected(true);
+    } else {
+      setRowsSelected(false);
+    }
+  };
+
+  const renderBulkActions = (selectedFlatRows) => {
+    if (rowsSelected) {
+      return ([
+        <RemindAction />,
+        <CancelAction />,
+        <ExtraAction />,
+      ]);
+    } else {
+      return ([]);
+    }
+  }
+  return (
+    <DataTable
+      isSortable
+      isSelectable
+      manualSortBy
+      isPaginated
+      manualPagination
+      isFilterable
+      manualFilters
+      isLoading={isLoading}
+      defaultColumnValues={{ Filter: TableTextFilter }}
+      FilterStatusComponent={FilterStatus}
+      columns={[
+        {
+          Header: 'Assignment details',
+          accessor: 'assignmentDetails',
+          Cell: AssignmentDetailsTableCell,
+          disableSortBy: true,
+        },
+        {
+          Header: 'Amount',
+          Cell: ({ row }) => `-${formatPrice(row.original.contentQuantity / 100, { maximumFractionDigits: 0 })}`,
+          disableFilters: true,
+        },
+      ]}
+      additionalColumns={[
+        {
+          id: 'action',
+          Header: '',
+          Cell: ({ row }) => (
```

Comment:
Should define `Cell` components as a standalone component, perhaps in its own file, similar to the comment about defining components inline:

```
Do not define components during render.
```

We should likely just pass the element name to the `Cell` property to avoid the pattern of defining components during render. Note: it will throw a prop type warning, though (should be fixed through [this issue](https://github.com/openedx/paragon/issues/2737) and [associated PR](https://github.com/openedx/paragon/pull/2749)).

Note x2: The `Amount` column above can define an inline function for its `Cell` attribute because it does not render JSX, so it's not technically a React component like this `action` column would be.